icetransport.go

```diff
@@ -157,6 +157,10 @@ func (t *ICETransport) Start(gatherer *ICEGatherer, params ICEParameters, role *
 		return err
```

Comment:
Can't comment on the line but it looks like you're `Lock`ing the `t.lock` above and never unlocking; is that on purpose?

Reply:
yeah there's a defer unlock above. not sure how i feel about that pattern!

Reply:
I don't like it haha, but since it was pre-existing and I have basically no context here will approve.